src/sql/src/plan/statement/dml.rs

```diff
@@ -64,7 +64,7 @@ pub fn plan_insert(
 ) -> Result<Plan, anyhow::Error> {
     let (id, mut expr) = query::plan_insert_query(scx, table_name, columns, source)?;
     expr.bind_parameters(&params)?;
-    let expr = expr.lower();
+    let expr = expr.optimize(&scx.into());
```

Comment:
If you add this here, it should also go into `plan_read_then_write`.

Reply:
I did it for the `selection` field in `plan_read_then_write`, I don't think I need to replace the `.lower_uncorrelated()` calls for the individual assignments, as the path for lowering `HirScalarExpr` instances that don't have nested `HirRelationExpr` elements will not be affected by the introduction of QGM.